src/main/java/io/cdap/plugin/topn/TopNConfig.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright © 2017-2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.topn;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.FailureCollector;
+
+import javax.annotation.Nullable;
+
+/**
+ * Config for top N of plugins.
+ */
+public class TopNConfig extends PluginConfig {
+
+  public static final int MAX_TOP = 1024*1024;
+  public static final String FIELD = "field";
+  public static final String SIZE = "size";
+  public static final String IGNORE_NULL = "ignoreNull";
+
+  @Name(FIELD)
+  @Description("The field which top results are sorted by.")
+  @Macro
+  private String field;
+
+  @Name(SIZE)
+  @Description("The maximum number of top records sorted by 'field' in result (DEFAULT: 1)")
+  @Macro
+  @Nullable
+  private Integer size;
+
+  @Name(IGNORE_NULL)
+  @Description("Set to 'true' to ignore records with null value in the field to sort by (DEFAULT : false)")
+  @Macro
+  @Nullable
+  private Boolean ignoreNull;
+
+  public TopNConfig() {
+    this.field = "";
+    this.size = null;
+    this.ignoreNull = null;
+  }
+
+  public void validate(FailureCollector failureCollector, @Nullable Schema inputSchema) {
+
+    // Check existence and type of the field.
+    if (!containsMacro(FIELD)) {
+      Schema.Field field = inputSchema.getField(getTopField());
+      if (field == null) {
+        failureCollector.addFailure(
+          String.format("Field '%s' does not exist in the input schema.", getTopField()),
```

Comment:
`Field '%s' must exist in the input schema.` and keep corrective action null
This is the pattern we are following for such error messages

Reply:
Done.